src/webgpu/web_platform/copyToTexture/canvas.spec.ts

```diff
@@ -135,18 +210,44 @@ class F extends CopyToTextureUtils {
     }
 
     // Generate expectedPixels
+    // Use getImageData and readPixels to get canvas contents.
     const rep = kTexelRepresentationInfo[format];
     const divide = 255.0;
+    let rgba: { R: number; G: number; B: number; A: number };
     for (let i = 0; i < height; ++i) {
       for (let j = 0; j < width; ++j) {
         const pixelPos = i * width + j;
+
+        rgba = {
+          R: sourcePixels[pixelPos * 4] / divide,
+          G: sourcePixels[pixelPos * 4 + 1] / divide,
+          B: sourcePixels[pixelPos * 4 + 2] / divide,
+          A: sourcePixels[pixelPos * 4 + 3] / divide,
+        };
+
+        if (!srcPremultiplied) {
+          if (dstPremultiplied) {
+            rgba.R *= rgba.A;
+            rgba.G *= rgba.A;
+            rgba.B *= rgba.A;
+          }
+        }
+
+        if (srcPremultiplied && !dstPremultiplied) {
+          if (rgba.A !== 0) {
```

Comment:
Would it work to assert this instead? Having this as an `if()` creates a corner case where I'm not sure the generated values are correct.